subprojects/dependency-lock/src/integTest/groovy/org/gradle/internal/dependencylock/generation/DependencyLockFileResolvedDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,437 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.dependencylock.generation
+
+import static org.gradle.internal.dependencylock.fixtures.DependencyLockFixture.*
+
+class DependencyLockFileResolvedDependenciesIntegrationTest extends AbstractDependencyLockFileGenerationIntegrationTest {
+
+    def "can create locks for dependencies with concrete version"() {
+        given:
+        mavenRepo.module('foo', 'bar', '1.5').publish()
+
+        buildFile << mavenRepository(mavenRepo)
+        buildFile << customConfigurations(MYCONF_CUSTOM_CONFIGURATION)
+        buildFile << """
+            dependencies {
+                myConf 'foo:bar:1.5'
+            }
+        """
+        buildFile << copyLibsTask(MYCONF_CUSTOM_CONFIGURATION)
+
+        when:
+        succeedsWithEnabledDependencyLocking(COPY_LIBS_TASK_NAME)
+
+        then:
+        lockFile.text == """{
+${commentAndLockFileVersionJson()}
+  "projects": [
+    {
+      "path": ":",
+      "configurations": [
+        {
+          "name": "myConf",
+          "dependencies": [
+            {
+              "moduleId": "foo:bar",
+              "requestedVersion": "1.5",
+              "lockedVersion": "1.5"
+            }
+          ]
+        }
+      ]
+    }
+  ]
+}"""
+        sha1File.text == '47ec5ad9e745cef18cea6adc42e4be3624572c9f'
+    }
+
+    def "can write locks if at least one dependency is resolvable"() {
```

Comment:
Maybe the title of the test case doesn't properly express the intent. What we are testing here is the following: We have two dependencies, one can be resolved, the other one cannot be resolved. Any dependencies that could be resolved are going to be written to the lock file even if the build fails due to a later, unresolved dependency.

Reply:
No, the title of the test is clear. But I can't think why we would want that behaviour. What's the point of creating a lock file containing a partial set of versions after a resolution failure? What's the use case that this would support.
Unless we have a good reason to do this, I think we should not create/update locks on resolution failure.

Reply:
My first intent was to make it beneficial for projects with large dependencies graphs here. If we already wrote locks for resolved dependencies then we wouldn't have to do that again and only do it for the ones that haven't been captured yet.

Thinking about it again from the perspective of consuming the lock file then I guess a half-captured resolution result could actually lead to confusing behavior for the user. As user you'd probably expect it to have locked all dependencies which is not the case.

I will change the code to only write the lock file if the project was successful.